server/simple_server.go

```diff
@@ -327,7 +338,7 @@ func ContextWithUserIP(ctx netContext.Context, r *http.Request) netContext.Conte
 	if err != nil {
 		LogWithFields(r).Warningf("unable to get IP: %s", err)
```

Comment:
Could we return here in case of err? Then we can drop that `else` block and just `return netContext.WithContext(...)`